app/views/admin/OutcomesReportView.coffee

```diff
@@ -192,66 +198,82 @@ OutcomesReportComponent = Vue.extend
       window.currentView = undefined
       application.router.openView(resultView)
 
+    fetchData: ->
+      @fetchByEmail().then (trimTeacher) =>
+        @fetchCompleteUser(trimTeacher).then (teacher) =>
+          @teacher = teacher
+          Promise.all([
+            @fetchTrialRequest(teacher).then (trialRequests) =>
+              @trialRequest = trialRequests[0]
+            @fetchClassrooms(teacher).then (classrooms) =>
+              @classrooms = classrooms
+              Promise.all @classrooms.map (classroom) =>
+                @fetchStudentSessions(classroom).then (sessions) =>
+                  Vue.set classroom, 'sessions', sessions
+            Promise.all([
+              @fetchCourseInstances(teacher).then (courseInstances) =>
+                @courseInstances = courseInstances
+              @fetchCourses().then (courses) =>
+                @courses = courses
+            ]).then ([courseInstances, courses]) =>
+              courseIDs = _.uniq courseInstances.map (courseInstance) =>
+                courseInstance.courseID
+              indexedCourses = _.indexBy(courses, '_id')
+              @courses = utils.sortCourses(courseIDs.map (courseID) =>
+                indexedCourses[courseID]
+              )
+          ])
```

Comment:
Generally prefer spacing the `.then`'s so that they are flat. So

```
@fetchByEmail()
.then (trimTeacher) =>
  @fetchCompleteUser(trimTeacher)
.then (teacher) =>
  @teacher = teacher
  ...
```

Though.. it's a bit less clear how to avoid nesting all the Promise.alls. Probably just further breaking logic out into separate functions.